aws/sdk/examples/sts/src/bin/credentials-provider.rs

```diff
@@ -3,91 +3,69 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-use aws_auth::{CredentialsError, ProvideCredentials};
-use std::sync::{Arc, Mutex};
-use std::time::{Duration, SystemTime};
+use aws_auth::refresh::{
+    AsyncCredentialLoader, AsyncCredentialResult, RefreshingCredentialProvider,
+};
+use aws_auth::CredentialsError;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::time::Duration;
 use sts::Credentials;
 
+#[derive(Clone)]
+struct StsCredentialLoader {
+    client: sts::Client,
+}
+
+impl AsyncCredentialLoader for StsCredentialLoader {
```

Comment:
we probably want something like `tower::service_fn` that accepts a "regular" async function so customers don't need to do the box/pinning themselves